### ISSUE TITLE
SET_OPCODE_OR_RETURN undefined if configured with --disable-gil 

### ISSUE THREAD

**Issue by smontanaro** [python/cpython#126513]
# Bug report

### Bug description:

The latest commit seems to have broken `specialized.c` when the GIL is disabled.

```
gcc -c -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall    -D_Py_TIER2=1 -D_Py_JIT -std=c11 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden  -I./Include/internal -I./Include/internal/mimalloc  -I. -I./Include    -DPy_BUILD_CORE -o Python/specialize.o Python/specialize.c
Python/specialize.c:2801:5: error: call to undeclared function 'SET_OPCODE_OR_RETURN'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 2801 |     SET_OPCODE_OR_RETURN(instr, CONTAINS_OP);
      |     ^
1 error generated.
```

Execute the basic `./configure` and compilation succeeds. Add `--disable-gil` to the configure command line and the above error is generated.

### CPython versions tested on:

CPython main branch

### Operating systems tested on:

Linux, macOS

### Linked PRs
* gh-126517

**Comment by Eclips4** [comment]
Thanks Skip for the report.
Confirmed on current main

**Comment by JelleZijlstra** [comment]
cc @mpage

**Comment by brettcannon** [comment]
In case anyone wants to see the failure in a buildbot, https://buildbot.python.org/all/#/builders/1046/builds/6656 .

**Comment by Eclips4** [comment]
Fixed in #126517.
